src/Rinds/OrRuleProcessor.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Rule processor that performs an OR operation on the rule's left and right
+ * operands.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\Rinds;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Rule processor that performs an OR operation on the rule's left and right
+ * operands.
+ */
+class OrRuleProcessor {
+	/**
+	 * Constructor.
+	 *
+	 * @param RuleEvaluator $rule_evaluator The rule evaluator to use.
+	 */
+	public function __construct( $rule_evaluator ) {
+		$this->rule_evaluator = $rule_evaluator;
+	}
+
+	/**
+	 * Performs an OR operation on the rule's left and right operands.
+	 *
+	 * @param object $rule The specific rule being processed by this rule processor.
+	 *
+	 * @return bool The result of the operation.
+	 */
+	public function process( $rule ) {
+		$evaluated_left_operand = $this->rule_evaluator->evaluate( $rule->left_operand );
```

Comment:
Does this mean we're limited to two operands with `OR` or is there a way to add more?

Reply:
Limited to two, I didn't really think about more than two operands as you could do that by nesting rules but I can see how that could get messy. The most flexible way would be an array of operands, is that what you're thinking?

Reply:
Ok, instead of `left_operand`/`right_operand` this now takes `operands` - an array of operands, each of which is evaluated by ANDing them together.